cmd/controller/run.go

```diff
@@ -73,12 +76,16 @@ func Run(opts Options, runLog logr.Logger) {
 	}
 
 	{ // add controller for apps
+		appRefTracker := reftracker.NewAppRefTracker()
+		appUpdateStatus := reftracker.NewAppUpdateStatus()
 		ctrlAppOpts := controller.Options{
 			Reconciler: NewUniqueReconciler(&ErrReconciler{
 				delegate: &AppsReconciler{
-					appClient:  appClient,
-					appFactory: appFactory,
-					log:        runLog.WithName("ar"),
+					appClient:       appClient,
+					appFactory:      appFactory,
+					log:             runLog.WithName("ar"),
+					AppRefTracker:   appRefTracker,
```

Comment:
this field should not be public.

Reply:
Changed to private and exposed via func for using in tests.